src/mobile/src/ui/views/wallet/StateExport.js

```diff
@@ -0,0 +1,198 @@
+        t: PropTypes.func.isRequired,
+        /** @ignore */
+        theme: PropTypes.object.isRequired,
+        /** @ignore */
+        accounts: PropTypes.object.isRequired,
+        /** @ignore */
+        settings: PropTypes.object.isRequired,
+        /** @ignore */
+        notificationLog: PropTypes.array.isRequired,
+        /** @ignore */
+        generateAlert: PropTypes.func.isRequired,
+    };
+
+    componentDidMount() {
+        leaveNavigationBreadcrumb('StateExport');
+    }
+
+    renderBackOption() {
+        const { theme, t } = this.props;
+
+        return (
+            <TouchableOpacity
+                onPress={() => this.props.setSetting('mainSettings')}
+                hitSlop={{ top: height / 55, bottom: height / 55, left: width / 55, right: width / 55 }}
+            >
+                <View style={styles.itemLeft}>
+                    <Icon name="chevronLeft" size={width / 28} color={theme.body.color} />
+                    <Text style={[styles.titleTextLeft, { color: theme.body.color }]}>{t('global:back')}</Text>
+                </View>
+            </TouchableOpacity>
+        );
+    }
+
+    renderExportOption() {
+        const { accounts, settings, notificationLog, t, theme } = this.props;
+
+        return (
+            <TouchableOpacity
+                onPress={async () => {
+                    if (isAndroid) {
+                        await getAndroidFileSystemPermissions();
+                    }
+
+                    const path = `${
+                        isAndroid ? RNFetchBlob.fs.dirs.DownloadDir : RNFetchBlob.fs.dirs.CacheDir
+                    }/Trinity-${moment().format('YYYYMMDD-HHmm')}.txt`;
+
+                    const fs = RNFetchBlob.fs;
+
+                    try {
+                        const files = await Promise.all([
+                            fs.ls(fs.dirs.DocumentDir),
+                            fs.ls(fs.dirs.CacheDir),
+                            fs.ls(fs.dirs.MainBundleDir),
+                        ]);
+
+                        const fileExists = await fs.exists(path);
+                        if (fileExists) {
+                            fs.unlink(path);
+                        }
+                        await fs.createFile(
+                            path,
+                            serialise(
+                                {
+                                    accounts: pick(accounts, ['onboardingComplete', 'accountInfo']),
+                                    settings: pick(settings, ['versions']),
+                                    notificationLog,
+                                    storageFiles: filter(flatMap(files), (name) => includes(name, 'realm')),
+                                },
+                                null,
+                                4,
+                            ),
+                        );
+                        this.props.generateAlert('success', t('exportSuccess'), t('exportSuccessExplanation'));
```

Comment:
We should trigger the share menu after writing the file, especially since iOS users do not generally have access to the filesystem (so they will not be able to access the file we just created). We can use a similar approach to the one we use for exporting a SeedVault:

https://github.com/iotaledger/trinity-wallet/blob/e3290d2333eb46d986336b90259168cda210cfe2/src/mobile/src/ui/components/SeedVaultExportComponent.js#L296-L307